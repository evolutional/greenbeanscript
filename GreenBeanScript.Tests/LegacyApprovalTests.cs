using System;
using System.IO;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using GreenBeanScript.Libs;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace GreenBeanScript.Tests
{
    [TestClass]
    [UseReporter(typeof (DiffReporter))]
    public class LegacyApprovalTests
    {
        private void Do(StringBuilder log, string libFileName)
        {
            Action<string> printFunc = (s) =>
            {
                log.AppendLine(s);
            };
            var stdLib = new StdLibrary(printFunc);
            var uut = new Machine(stdLib);
            var newLib = new Library();
            newLib.LoadFromFile(uut, libFileName);

            uut.ExecuteLibrary(newLib);
        }

        private void List(StringBuilder log, string libFileName)
        {
            var newLib = new Library();
            newLib.ListLibraryFromFile(log, libFileName);
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledHeapsort()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/heapsort.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledFib()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/fib.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledMatrix()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/matrix.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledHash()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/hash.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledFloat()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/float.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledFib()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/fib.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledMatrix()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/matrix.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledHeapsort()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/heapsort.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledHash()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/hash.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledFloat()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/float.gml");
            Approvals.Verify(log.ToString());
        }
    }
}
