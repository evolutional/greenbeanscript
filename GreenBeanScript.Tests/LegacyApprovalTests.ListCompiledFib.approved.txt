GM LIBRARY C:\dev\greenbeanscript\GreenBeanScript.Tests\ExampleScripts\fib.gml, DEBUG=0

STRING TABLE, SIZE=   22
===============================================================================
00000:print
00006:*** FIB ***
00018:fib

NO SOURCE CODE IN LIB
FUNCTIONS, COUNT=2
===============================================================================

===============================================================================
FUNCTION ID 0
FLAGS 0
NUM PARAMS 1
NUM LOCALS 0
MAX STACK 6
BYTE CODE SIZE 124
  0000 get local 0
  0008 push int 2
  0016 lt
  0020 brz 36
  0028 push int 1
  0032 retv
  0036 push null
  0040 get global 18
  0048 get local 0
  0056 push int 2
  0064 sub
  0068 call 1
  0076 push null
  0080 get global 18
  0088 get local 0
  0096 push int 1
  0100 sub
  0104 call 1
  0112 add
  0116 retv
  0120 ret


===============================================================================
FUNCTION ID 1
FLAGS 1
NUM PARAMS 0
NUM LOCALS 1
MAX STACK 5
BYTE CODE SIZE 120
  0000 push null
  0004 get global 0
  0012 push str 6
  0020 call 1
  0028 pop
  0032 push fn 0
  0040 set global 18
  0048 push int 32
  0056 set local 0
  0064 push null
  0068 get global 0
  0076 push null
  0080 get global 18
  0088 get local 0
  0096 call 1
  0104 call 1
  0112 pop
  0116 ret


END LIB
